### ISSUE TITLE
jax.scipy.special.i0 returns nonzero gradient at zero

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#40628]
### Description

`jax.scipy.special.i0` appears to return an incorrect nonzero first-order derivative at zero.

`I0` is an even entire function, so:

```
I0'(0) = 0
```

In the example below, the differentiated function is `i0(2 * t - 1)`. At `t = 0.5`, the inner argument is zero, so the first derivative with respect to `t` should also be `0.0`. However, JAX returns `2.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "")
os.environ.setdefault("OMP_NUM_THREADS", "1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import jax
    import jax.numpy as jnp
    import jax.scipy.special
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

def target(t):
    return jax.scipy.special.i0(2 * t - 1)

x = jnp.asarray(0.5, dtype=jnp.float64)
expected = 0.0

actual = float(jax.grad(target)(x))

print("actual:", actual)
print("expected:", expected)

if not (math.isfinite(actual) and abs(actual - expected) <= 1e-12):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
actual: 2.0
expected: 0.0
```

### Expected result

```
0.0
```

The forward value is correct, but the first derivative is not. Since `I0` is even, its derivative at zero should be exactly zero. JAX returns `2.0`, suggesting that the autodiff rule for `jax.scipy.special.i0` is incorrect at the zero point.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
Thanks for the report. A simpler reproducer would make the issue clearer; for example:
```python
>>> import jax
>>> jax.grad(jax.scipy.special.i0)(0.0)
Array(1.0000001, dtype=float32, weak_type=True)
```
I suspect this comes from the fact that `grad(abs)(0.0)` returns `1.0`, as `abs` is used in the implementation of `i0` here: https://github.com/jax-ml/jax/blob/0d00c5e3471d70afa871f96a0e125b118bcef1b8/jax/_src/scipy/special.py#L1806

**Comment by jakevdp** [comment]
One thing to note, `jax.numpy.i0` doesn't have this problem:
```python
>>> import jax
>>> jax.grad(jax.numpy.i0)(0.0)
Array(0., dtype=float32, weak_type=True)
```
The `jax.numpy` version defines a custom JVP rule; we should probably do the same for the scipy version.
